### PR TITLE
Feature/xml descriptions gui sc2

### PR DESCRIPTION
**Pull request by DawidSA2017**

### CODE REVIEW

**Comment by DawidSA2017** [comment]
Hello, I've fixed the merge conflicts to your tp17_team1, you are free to review!

File: plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/xml/XmlFeatureModelFormat.java [hunk 1]
```diff
@@ -479,6 +473,10 @@ private void parseConstraints(NodeList nodeList) throws UnsupportedModelExceptio
 					throwError("Feature \"" + featureName + "\" does not exists", e);
 				}
 			} else if (nodeName.equals(DESCRIPTION)) {
+				// The method should return without adding any nodes,
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/xml/XmlFeatureModelFormat.java]: You should change the comment-style here.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/ConstraintDialog.java [hunk 2]
```diff
@@ -744,38 +761,59 @@ public void modifyText(ModifyEvent e) {
...
+		final GridLayout featureGroupLayout = new GridLayout();
+		featureGroupLayout.numColumns = 1;
+		descriptionGroup.setLayout(featureGroupLayout);
+
+		constraintDescriptionText = new Text(descriptionGroup, SWT.WRAP | SWT.BORDER | SWT.MULTI | SWT.V_SCROLL);
+		constraintDescriptionText.setLayoutData(gridData);
+		constraintDescriptionText.setText(description);
+	}
+
+	/**
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/ConstraintDialog.java:784]: This is redundant

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/figures/ConstraintFigure.java [hunk 3]
```diff
@@ -168,6 +168,16 @@ public void setConstraintProperties() {
...
+		
+		final String description = constraint.getDescription();
+		if ((description != null) && !description.trim().isEmpty()) {
+			toolTipContent.add(new Label("Description:"));
+			toolTipContent.add(new Label(description));
+		} else {
+			toolTipContent.add(new Label("No Description"));
+		}
+		
+
```
Comment (Henningson) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/figures/ConstraintFigure.java]: Too many empty lines

**Pull request merged**